terraform/environments/digital-prison-reporting/main.tf

```diff
@@ -250,6 +251,227 @@ module "glue_hive_table_creation_job" {
   ]
 }
 
+# Glue Job, Switch Prisons Hive Data Location
+module "glue_switch_prisons_hive_data_location_job" {
+  source                        = "./modules/glue_job"
+  create_job                    = local.create_job
+  name                          = "${local.project}-switch-prisons-hive-data-location-${local.env}"
+  short_name                    = "${local.project}-switch-prisons-hive-data-location"
+  command_type                  = "glueetl"
+  description                   = "Switch Prisons Hive tables data location"
+  create_security_configuration = local.create_sec_conf
+  job_language                  = "scala"
+  temp_dir                      = "s3://${module.s3_glue_job_bucket.bucket_id}/tmp/${local.project}-switch-prisons-hive-data-location-${local.env}/"
+  spark_event_logs              = "s3://${module.s3_glue_job_bucket.bucket_id}/spark-logs/${local.project}-switch-prisons-hive-data-location-${local.env}/"
+  # Placeholder Script Location
+  script_location              = local.glue_placeholder_script_location
+  enable_continuous_log_filter = false
+  project_id                   = local.project
+  aws_kms_key                  = local.s3_kms_arn
+
+  execution_class             = "STANDARD"
+  worker_type                 = "G.1X"
+  number_of_workers           = 2
+  max_concurrent              = 64
+  region                      = local.account_region
+  account                     = local.account_id
+  log_group_retention_in_days = 1
+
+  tags = merge(
+    local.all_tags,
+    {
+      Name          = "${local.project}-switch-prisons-hive-data-location-${local.env}"
+      Resource_Type = "Glue Job"
+      Jira          = "DPR2-46"
+    }
+  )
+
+  arguments = {
+    "--extra-jars"                             = local.glue_jobs_latest_jar_location
+    "--class"                                  = "uk.gov.justice.digital.job.SwitchHiveTableJob"
+    "--dpr.aws.region"                         = local.account_region
+#    "--dpr.config.key"                         = "(Required) config key. Will be specified at point of call in the reload step-functions"
+    "--dpr.prisons.data.switch.target.s3.path" = "s3://${module.s3_temp_reload_bucket.bucket_id}"
+    "--dpr.prisons.database"                   = module.glue_prisons_database.db_name
+    "--dpr.contract.registryName"              = module.s3_schema_registry_bucket.bucket_id
+    "--dpr.schema.cache.max.size"              = local.hive_table_creation_job_schema_cache_max_size
+    "--dpr.log.level"                          = local.refresh_job_log_level
+  }
+
+  depends_on = [
+    module.s3_raw_archive_bucket.bucket_id,
+    module.s3_structured_bucket.bucket_id,
+    module.s3_curated_bucket.bucket_id,
+    module.glue_raw_archive_database.db_name,
+    module.glue_structured_zone_database.db_name,
+    module.glue_curated_zone_database.db_name,
+    module.glue_prisons_database.db_name,
+    module.glue_registry_avro.registry_name
+  ]
+}
+
+# Glue Job, S3 File Archive Job
```

Comment:
This glue job replaces `s3_file_transfer_lambda` which will be removed in a later PR.